tests/third_party/cupy/linalg_tests/test_decomposition.py

```diff
@@ -135,3 +136,74 @@ def check_L(self, array):
     def test_decomposition(self, dtype):
         A = numpy.array([[1, -2], [-2, 1]]).astype(dtype)
         self.check_L(A)
+
+
+@testing.parameterize(
+    *testing.product(
+        {
+            "mode": ["r", "raw", "complete", "reduced"],
+        }
+    )
+)
+class TestQRDecomposition(unittest.TestCase):
+    @testing.for_dtypes("fdFD")
+    def check_mode(self, array, mode, dtype):
+        if dtype in (numpy.complex64, numpy.complex128):
```

Comment:
I believe you should implement `ungqr` and `ungqr_batch` rather than skipping the tests

Reply:
I really overlooked that point
I added `ungrf` and `ungqr_batch` implementation in the new commits